### ISSUE TITLE
Repeating results while training

### ISSUE THREAD

**Issue by lincolnBush** [apache/incubator-mxnet#5273]
Dear all:

I was running a image regression problem based on these code:

train = mx.io.NDArrayIter(X_train, y_train, batch_size, shuffle=True) #, label_name='lin_reg_label')
val = mx.io.NDArrayIter(X_test, y_test, batch_size)

#Lenet Symbol Configuration
#data = mx.sym.Variable('data')
#net_label = mx.sym.Variable('softmax_label')
# first conv
#conv1 = mx.symbol.Convolution(data=data, kernel=(3,3), num_filter=20)
#tanh1 = mx.symbol.Activation(data=conv1, act_type="tanh")
#pool1 = mx.symbol.Pooling(data=tanh1, pool_type="max",kernel=(2,2), stride=(2,2))
# second conv
#conv2 = mx.symbol.Convolution(data=pool1, kernel=(5,5), num_filter=50)
#tanh2 = mx.symbol.Activation(data=conv2, act_type="tanh")
#pool2 = mx.symbol.Pooling(data=tanh2, pool_type="max",kernel=(2,2), stride=(2,2))
# first fullc
#flatten = mx.symbol.Flatten(data=pool1)
#fc1 = mx.symbol.FullyConnected(data=flatten, num_hidden=100)
#tanh3 = mx.symbol.Activation(data=fc1, act_type="tanh")
#second fullc
#fc2 = mx.symbol.FullyConnected(data=tanh3, num_hidden=1)
# loss
#lenet = mx.symbol.SoftmaxOutput(data=fc2, name='softmax')
#loss = mx.symbol.MAERegressionOutput(data=fc2, label=net_label, name='loss')

net_in = mx.symbol.Variable('data')
net_in_flat = mx.symbol.Flatten(data=net_in)
net_label = mx.symbol.Variable('softmax_label')
fc1 = mx.symbol.FullyConnected(data=net_in_flat, name='fc1', num_hidden=512)
fc1_act = mx.symbol.Activation(data=fc1, name='fc1_act', act_type='sigmoid')
fc2 = mx.symbol.FullyConnected(data=fc1_act, name='fc2', num_hidden=256)
fc2_act = mx.symbol.Activation(data=fc2, name='fc2_act', act_type='sigmoid')
fc3 = mx.symbol.FullyConnected(data=fc2_act, name='fc3', num_hidden=256)
#net_out = mx.symbol.FullyConnected(data=fc3, name='net_out', num_hidden=1)
loss = mx.symbol.LogisticRegressionOutput(data=fc3, label=net_label, name='loss')
  
logging.basicConfig(level=logging.INFO)
logger = logging.getLogger(__name__) 

model = mx.module.Module(
    symbol=lenet,
    data_names=['data'],
    label_names=['softmax_label'],
    logger = logger)

model.bind(data_shapes=train.provide_data, label_shapes=train.provide_label)

model.init_params(initializer=mx.init.Xavier(rnd_type='gaussian', factor_type="in", magnitude=2))

model.fit(
    train_data=train, 
    eval_data=val,
    eval_metric='rmse', 
    optimizer='sgd', 
    optimizer_params=(('learning_rate', 0.001), ('momentum', 0.5)),
    num_epoch = train_epoch,
    batch_end_callback=mx.callback.Speedometer(batch_size, frequent=100)
)

The problem is that the result in Epoch[0] and Epoch[1] are the same. What could cause this problem. Many thanks.

WARNING:__main__:Already binded, ignoring bind()
INFO:root:Epoch[0] Batch [100]	Speed: 853.11 samples/sec	Train-rmse=0.954174
INFO:root:Epoch[0] Batch [200]	Speed: 918.76 samples/sec	Train-rmse=0.953405
INFO:root:Epoch[0] Batch [300]	Speed: 904.48 samples/sec	Train-rmse=0.954708
INFO:root:Epoch[0] Batch [400]	Speed: 900.66 samples/sec	Train-rmse=0.951832
INFO:root:Epoch[0] Batch [500]	Speed: 875.52 samples/sec	Train-rmse=0.954197
INFO:root:Epoch[0] Batch [600]	Speed: 822.84 samples/sec	Train-rmse=0.953868
INFO:root:Epoch[0] Batch [700]	Speed: 868.63 samples/sec	Train-rmse=0.953722
INFO:root:Epoch[0] Batch [800]	Speed: 882.28 samples/sec	Train-rmse=0.954360
INFO:root:Epoch[0] Batch [900]	Speed: 871.47 samples/sec	Train-rmse=0.953438
INFO:root:Epoch[0] Batch [1000]	Speed: 888.41 samples/sec	Train-rmse=0.952074
INFO:root:Epoch[0] Batch [1100]	Speed: 876.72 samples/sec	Train-rmse=0.954426
INFO:root:Epoch[0] Batch [1200]	Speed: 872.66 samples/sec	Train-rmse=0.954826
INFO:root:Epoch[0] Batch [1300]	Speed: 855.40 samples/sec	Train-rmse=0.952572
INFO:root:Epoch[0] Batch [1400]	Speed: 823.90 samples/sec	Train-rmse=0.956657
INFO:root:Epoch[0] Batch [1500]	Speed: 843.67 samples/sec	Train-rmse=0.952269
INFO:root:Epoch[0] Batch [1600]	Speed: 853.12 samples/sec	Train-rmse=0.952531
INFO:root:Epoch[0] Batch [1700]	Speed: 864.87 samples/sec	Train-rmse=0.953331
INFO:root:Epoch[0] Batch [1800]	Speed: 861.15 samples/sec	Train-rmse=0.952314
INFO:root:Epoch[0] Batch [1900]	Speed: 857.00 samples/sec	Train-rmse=0.954161
INFO:root:Epoch[0] Batch [2000]	Speed: 863.47 samples/sec	Train-rmse=0.953831
INFO:root:Epoch[0] Batch [2100]	Speed: 842.78 samples/sec	Train-rmse=0.954865
INFO:root:Epoch[0] Batch [2200]	Speed: 856.31 samples/sec	Train-rmse=0.955545
INFO:root:Epoch[0] Batch [2300]	Speed: 853.57 samples/sec	Train-rmse=0.951860
INFO:root:Epoch[0] Batch [2400]	Speed: 853.34 samples/sec	Train-rmse=0.952450
INFO:__main__:Epoch[0] Train-rmse=0.953217
INFO:__main__:Epoch[0] Time cost=358.616
INFO:__main__:Epoch[0] Validation-rmse=0.907619
INFO:root:Epoch[1] Batch [100]	Speed: 885.21 samples/sec	Train-rmse=0.954174
INFO:root:Epoch[1] Batch [200]	Speed: 855.40 samples/sec	Train-rmse=0.953405
INFO:root:Epoch[1] Batch [300]	Speed: 835.95 samples/sec	Train-rmse=0.954708
INFO:root:Epoch[1] Batch [400]	Speed: 847.92 samples/sec	Train-rmse=0.951832
INFO:root:Epoch[1] Batch [500]	Speed: 857.69 samples/sec	Train-rmse=0.954197
INFO:root:Epoch[1] Batch [600]	Speed: 856.08 samples/sec	Train-rmse=0.953868
INFO:root:Epoch[1] Batch [700]	Speed: 843.45 samples/sec	Train-rmse=0.953722
INFO:root:Epoch[1] Batch [800]	Speed: 852.43 samples/sec	Train-rmse=0.954360
INFO:root:Epoch[1] Batch [900]	Speed: 848.82 samples/sec	Train-rmse=0.953438
INFO:root:Epoch[1] Batch [1000]	Speed: 836.39 samples/sec	Train-rmse=0.952074
INFO:root:Epoch[1] Batch [1100]	Speed: 845.90 samples/sec	Train-rmse=0.954426
INFO:root:Epoch[1] Batch [1200]	Speed: 824.33 samples/sec	Train-rmse=0.954826
INFO:root:Epoch[1] Batch [1300]	Speed: 845.67 samples/sec	Train-rmse=0.952572
INFO:root:Epoch[1] Batch [1400]	Speed: 845.23 samples/sec	Train-rmse=0.956657
INFO:root:Epoch[1] Batch [1500]	Speed: 852.43 samples/sec	Train-rmse=0.952269
INFO:root:Epoch[1] Batch [1600]	Speed: 845.90 samples/sec	Train-rmse=0.952531
INFO:root:Epoch[1] Batch [1700]	Speed: 812.19 samples/sec	Train-rmse=0.953331
INFO:root:Epoch[1] Batch [1800]	Speed: 868.63 samples/sec	Train-rmse=0.952314
INFO:root:Epoch[1] Batch [1900]	Speed: 854.48 samples/sec	Train-rmse=0.954161
INFO:root:Epoch[1] Batch [2000]	Speed: 814.05 samples/sec	Train-rmse=0.953831
INFO:root:Epoch[1] Batch [2100]	Speed: 881.55 samples/sec	Train-rmse=0.954865
INFO:root:Epoch[1] Batch [2200]	Speed: 899.14 samples/sec	Train-rmse=0.955545
INFO:root:Epoch[1] Batch [2300]	Speed: 888.65 samples/sec	Train-rmse=0.951860
INFO:root:Epoch[1] Batch [2400]	Speed: 845.23 samples/sec	Train-rmse=0.952450
INFO:__main__:Epoch[1] Train-rmse=0.953217
INFO:__main__:Epoch[1] Time cost=364.212
INFO:__main__:Epoch[1] Validation-rmse=0.907619

**Comment by szha** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to ping me to reopen if this is still an active issue. Thanks!
